protocol/src/frame.rs

```diff
@@ -93,13 +95,38 @@ impl Frame {
                 .map_err(ProtocolError::SerdeError)?,
             Frame::Error(payload) => bincode::serialize_into(dst.writer(), &payload)
                 .map_err(ProtocolError::SerdeError)?,
-            Frame::BatchMessage(bytes) => dst.extend_from_slice(&bytes),
+            Frame::BatchMessage(payload) => bincode::serialize_into(dst.writer(), &payload)
+                .map_err(ProtocolError::SerdeError)?,
             Frame::Ok => (),
         }
 
         Ok(())
     }
 
+    pub fn unwrap_retention_policy(&self) -> u64 {
+        match self {
+            Self::RegisterPublisher(payload) => payload.retention_policy,
+            Self::RegisterSubscriber(payload) => payload.retention_policy,
+            _ => panic!("Attempted to unwrap non-register Frame variant"),
+        }
+    }
+
+    pub fn unwrap_batch_size(&self) -> u32 {
```

Comment:
I get that these don't have to take ownership, but maybe they should be called something else, or take ownership for symmetry?